src/data/real/ennreal.lean

```diff
@@ -1218,6 +1232,8 @@ begin
   exact nnreal.half_lt_self hz
 end
 
+lemma half_le_self : a / 2 ≤ a := le_add_self.trans_eq (add_halves _)
```

Comment:
I *think* you can get this one along with `ennreal.add_halves` for free if you reduce the assumptions on `add_halves`.

Reply:
What exactly do you suggest?

Reply:
Note that `ennreal` is not a `group_with_zero`, so theorems about `division_ring` etc do not apply.

Reply:
Feels like we're missing `division_semiring` here. Division seems to assume additive inverses, but I can't really see why they would be necessary for most things.

Reply:
It's not even a `group_with_zero` because of infinity, so `division_semiring` won't help. We'll add `division_semiring` once we have more than a couple of examples.